src/main/java/com/mraof/minestuck/entry/EntryProcess.java

```diff
@@ -40,104 +49,171 @@
+		{
+			player.sendSystemMessage(Component.literal("Entry not permitted from this dimension."));
+			return;
+		}
+		
+		if(!TitleSelectionHook.performEntryCheck(player))
+			return;
+		if(waitingProcess != null)
+		{
+			player.sendSystemMessage(Component.literal("Someone else is already entering."));
+			return;
+		}
+		
+		PlayerIdentifier identifier = IdentifierHandler.encode(player);
+		Optional<SburbConnection> c = SkaianetHandler.get(player.level).getPrimaryConnection(identifier, true);
+		
+		if(c.isPresent() && c.get().hasEntered())
+		{
+			secondEntryTeleport(player, c.get().getClientDimension());
+			return;
+		}
+		
+		ResourceKey<Level> landDimension = SkaianetHandler.get(player.level).prepareEntry(identifier);
+		if(landDimension == null)
+		{
+			player.sendSystemMessage(Component.literal("Something went wrong while creating your Land. More details in the server console."));
+			return;
+		}
+		
+		ServerLevel landLevel = Objects.requireNonNull(player.getServer()).getLevel(landDimension);
+		if(landLevel == null)
+		{
+			player.sendSystemMessage(Component.literal("Unable to find land dimension. Something is not as it should be!"));
+			return;
+		}
+		
+		EntryProcess process = new EntryProcess(player, landLevel);
+		if(!process.canModifyEntryBlocks(player.level, player))
+		{
+			player.sendSystemMessage(Component.literal("You are not allowed to enter here."));    //TODO translation key
+			return;
+		}
+		
+		landLevel.getChunkSource().addRegionTicket(CHUNK_TICKET_TYPE, new ChunkPos(0, 0), 0, Unit.INSTANCE);
+		
+		waitingProcess = process;
+		startTime = player.level.getGameTime() + MinestuckConfig.COMMON.entryDelay.get();
+		MSPacketHandler.sendToAll(new EntryEffectPackets.Effect(player.level.dimension(), process.origin, process.artifactRange));
+		LOGGER.info("Entry prep done in {}ms", System.currentTimeMillis() - time);
+	}
 	
-	private final int xDiff, yDiff, zDiff;
-	private final BlockPos origin;
-	private final boolean creative;
+	private static void secondEntryTeleport(ServerPlayer player, ResourceKey<Level> land)
+	{
+		if(MinestuckConfig.SERVER.stopSecondEntry.get())
+		{
+			player.sendSystemMessage(Component.literal("You have already entered, and are not permitted to re-enter."));
+			return;
+		}
+		if(MSDimensions.isLandDimension(player.server, player.level.dimension()))
+		{
+			player.sendSystemMessage(Component.literal("You may not re-enter from this dimension."));
+			return;
+		}
+		
+		ServerLevel landWorld = Objects.requireNonNull(player.getServer()).getLevel(land);
+		if(landWorld == null)
+		{
+			player.sendSystemMessage(Component.literal("Unable to find land dimension. Something is not as it should be!"));
+			return;
+		}
+		
+		//Teleports the player to their home in the Medium, without any bells or whistles.
+		BlockPos pos = new BlockPos(0, 100, 0);
+		Teleport.teleportEntity(player, landWorld, pos.getX() + 0.5F, pos.getY(), pos.getZ() + 0.5F, player.getYRot(), player.getXRot());
```

Comment:
In testing, its likely the player will reenter underground

Reply:
Has now been changed to use the heightmap for motion blocking blocks.